src/test/java/com/wikia/webdriver/common/core/elemnt/Wait.java

```diff
@@ -48,19 +48,15 @@ public WebElement forElementPresent(By by) {
    * Checks if the element is present in browser DOM
    */
   public WebElement forElementPresent(By by, boolean failOnTimeout) {
-    changeImplicitWait(250, TimeUnit.MILLISECONDS);
-    try {
-      return wait.until(ExpectedConditions.presenceOfElementLocated(by));
-    } catch (TimeoutException e) {
-      if (failOnTimeout) {
-        PageObjectLogging.log(ELEMENT_PRESENT_MESSAGE,
-                              String.format(ELEMENT_PRESENT_ERROR_FORMAT, by.toString()), false);
-      }
+    return forElementPresent(this.wait, by, failOnTimeout);
+  }
 
-      throw e;
-    } finally {
-      restoreDeaultImplicitWait();
-    }
+  /**
+   * Checks if the element is present in browser DOM
+   */
+  public WebElement forElementPresent(By by, boolean failOnTimeout, int timeout) {
+    WebDriverWait wait = new WebDriverWait(this.driver, timeout);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "wait" which hides the field declared at line 32. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AHiddenFieldCheck)